test/Airdrop.test.ts

```diff
@@ -45,45 +47,211 @@ describe("Merkle Airdrop Token", () => {
             });
         }
         badTree = generateTree(badRecepients);
+    });
+
+    beforeEach(async () => {
+        const Token = await ethers.getContractFactory("TokenXP");
+        token = await Token.connect(sender).deploy();
+        await token.connect(sender).deployed();
+
+        console.log("Token deployed at: ", token.address);
 
         const Airdrop = await ethers.getContractFactory("MerkleAirdrop");
-        airdrop = await Airdrop.deploy(merkleRoot);
+        airdrop = await Airdrop.deploy();
         await airdrop.deployed();
 
         console.log("Merkle Airdrop Token deployed at:", airdrop.address);
     });
 
+    describe("Function initialize", () => {
+        it("should not let other than owner to initialize", async () => {
+            await expect(
+                airdrop
+                    .connect(sender)
+                    .initialize(sender.address, token.address, merkleRoot)
+            ).to.be.revertedWithCustomError(airdrop, "NotOwner");
+        });
+
+        it("should revert if initialized with zero merkle root", async () => {
+            let zeroMerkleRoot =
+                "0x0000000000000000000000000000000000000000000000000000000000000000";
+            await expect(
```

Comment:
use `ethers.constants.AddressZero`

Reply:
are address(0) and bytes(0) same?

Reply:
sorry..it will be `ethers.constants.HashZero